bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/protocol/MPDConnectionThread.java

```diff
@@ -0,0 +1,283 @@
+    private AtomicBoolean isInIdle = new AtomicBoolean(false);
+    private AtomicBoolean disposed = new AtomicBoolean(false);
+
+    public MPDConnectionThread(MPDResponseListener listener, String address, Integer port, String password) {
+        this.listener = listener;
+
+        this.address = address;
+        this.port = port;
+        this.password = password;
+    }
+
+    @Override
+    public void run() {
+
+        try {
+            while (!disposed.get()) {
+                try {
+                    synchronized (pendingCommands) {
+                        pendingCommands.clear();
+                        pendingCommands.add(new MPDCommand("status"));
+                        pendingCommands.add(new MPDCommand("currentsong"));
+                    }
+
+                    establishConnection();
+                    updateThingStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, null);
+
+                    processPendingCommands();
+                } catch (IOException e) {
+                    updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                } catch (MPDException e) {
+                    updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+                }
+
+                isInIdle.set(false);
+                closeSocket();
+
+                if (!disposed.get()) {
+                    sleep(RECONNECTION_TIMEOUT);
+                }
+            }
+        } catch (InterruptedException e) {
+        }
+    }
+
+    /**
+     * dispose the connection
+     */
+    public void dispose() {
+        disposed.set(true);
+        synchronized (pendingCommands) {
+            pendingCommands.add(new MPDCommand("close"));
+            sendNoIdleIfInIdle();
+            interrupt();
+        }
+    }
+
+    /**
+     * add a command to the pending commands queue
+     *
+     * @param command command to add
+     */
+    public void addCommand(MPDCommand command) {
+        insertCommand(command, -1);
+    }
+
+    private void insertCommand(MPDCommand command, int position) {
+        logger.debug("insert command '{}' at position {}", command.getCommand(), position);
+        int index = position;
+        synchronized (pendingCommands) {
+            if (index < 0) {
+                index = pendingCommands.size();
+            }
+            pendingCommands.add(index, command);
+            sendNoIdleIfInIdle();
+        }
+    }
+
+    private void updateThingStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        if (!disposed.get()) {
+            listener.updateThingStatus(status, statusDetail, description);
+        }
+    }
+
+    private void sendNoIdleIfInIdle() {
+        if (isInIdle.compareAndSet(true, false)) {
+            try {
+                sendCommand(new MPDCommand("noidle"));
+            } catch (IOException e) {
```

Comment:
Do you ignore the exception by intention? Is it worth to log it?

Reply:
I have added a log, since it might fail.